aquadoggo/src/graphql/schema_builders/document.rs

```diff
@@ -0,0 +1,474 @@
+        ))
+        .argument(InputValue::new(
+            DOCUMENT_VIEW_ID_ARG,
+            TypeRef::named(DOCUMENT_VIEW_ID_SCALAR),
+        ))
+        .description(format!(
+            "Query a {} document by id or view id.",
+            schema.name()
+        )),
+    )
+}
+
+/// Add query for getting all documents of a certain schema type to the root query object.
+///
+/// Constructs an endpoint with the format `all_<SCHEMA_ID>`.
+pub fn build_all_document_query(query: Object, schema: &Schema) -> Object {
+    let schema_id = schema.id().clone();
+    query.field(
+        Field::new(
+            format!("{QUERY_ALL_PREFIX}{}", schema_id),
+            TypeRef::named_list(schema_id.to_string()),
+            move |ctx| {
+                let schema_id = schema_id.clone();
+                FieldFuture::new(async move {
+                    debug!("Query to {QUERY_ALL_PREFIX}{} received", schema_id);
+
+                    // Access the store.
+                    let store = ctx.data_unchecked::<SqlStore>();
+
+                    // Fetch all documents of the schema this endpoint serves and compose the
+                    // field value (a list) which will bubble up the query tree.
+                    //
+                    // TODO: Optimize with data loader.
+                    let documents: Vec<FieldValue> = store
+                        .get_documents_by_schema(&schema_id)
+                        .await?
+                        .iter()
+                        .map(|document| {
+                            FieldValue::owned_any((
+                                Some(DocumentIdScalar::from(document.id())),
+                                None::<DocumentViewIdScalar>,
+                            ))
+                        })
+                        .collect();
+
+                    // Pass the list up to the children query fields.
+                    Ok(Some(FieldValue::list(documents)))
+                })
+            },
+        )
+        .description(format!("Get all {} documents.", schema.name())),
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use async_graphql::{value, Response, Value};
+    use p2panda_rs::document::traits::AsDocument;
+    use p2panda_rs::identity::KeyPair;
+    use p2panda_rs::schema::FieldType;
+    use p2panda_rs::storage_provider::traits::DocumentStore;
+    use p2panda_rs::test_utils::fixtures::random_key_pair;
+    use rstest::rstest;
+    use serde_json::json;
+    use serial_test::serial;
+
+    use crate::test_utils::{add_document, add_schema, graphql_test_client, test_runner, TestNode};
+
+    #[rstest]
+    // Note: This and more tests in this file use the underlying static schema provider which is a
+    // static mutable data store, accessible across all test runner threads in parallel mode. To
+    // prevent overwriting data across threads we have to run this test in serial.
+    //
+    // Read more: https://users.rust-lang.org/t/static-mutables-in-tests/49321
+    #[serial]
```

Comment:
Did you try removing the `#[serial]` attributes? I think the tests should be able to run now concurrently as far as I remember the problem?

Reply:
Yep, did this 👍